docs/dev/managing-dependencies.md

```diff
@@ -0,0 +1,105 @@
+2. This dependency has had its version changed (managed) from the version defined in the parent transitive dependency's POM to the version for this dependency given in the `dependencyManagement` section of our POM. The "version managed from" in the brackets is the version which would have been used if it hadn't been overridden by specifying a managed version.
+3. This dependency was omitted because it was already resolved (found) earlier. If a dependency with different versions is seen more than once, subsequent instances are ignored as **conflicts**. In this case, the conflicting version is a transitive dependency and the resolved dependency is a direct dependency (declared in the module POM) which takes priority.
+
+This verbose output is also useful for discovering which dependencies need to have exclusions added when using [Maven dependency exclusions](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html#dependency-exclusions). As an example, replacing Log4j with Reload4j can be done be excluding all Log4j dependencies and adding Reload4j as a project dependency (from a Java perspective this works because both use the same package name). This requires an exclusion section to be added for every dependency which includes Log4j as a transitive dependency.
+
+Without the verbose output, only a single resolved Log4j dependency is shown. In a project with 5 dependencies with Log4j as a sub-dependency, if only the resolved dependency were to be excluded, then another would become the resolved dependency. With this approach, the tree would need to be printed repeatedly until all these dependencies had been seen and excluded. With the verbose output these dependencies are all shown initially.
+
+For more info, [see the Maven dependency plugin documentation for the 'tree' goal](https://maven.apache.org/plugins/maven-dependency-plugin/tree-mojo.html).
+
+### Output to File
+
+Unlike `dependency:list`, the `tree` plugin goal does output to file correctly. This puts all tree output into a single file called `mvn-dep-tree.txt` in the current directory:
+```
+mvn dependency:tree -DoutputFile=$(pwd)/mvn-dep-tree.txt -DappendOutput=true
+```
+
+## Plotting graphs of dependencies
+
+### Using the built-in Maven plugin
+
+In addition to text output, this can also produce `.dot` files which can be used to create dependency graph visualisations (in combination with [Graphviz](https://en.wikipedia.org/wiki/Graphviz)). The resulting files are simple and lack positioning information. They only work well if the number of dependencies is low (e.g. `-Dverbose=true` not used).
+
+This visualisation was produced using the plugin and [Graphviz `sfdp`](https://graphviz.org/docs/layouts/sfdp/).
+
+```bash
+mvn dependency:tree -DoutputType=dot -DoutputFile=$(pwd)/core.dot -pl :core # Create the .dot file
```

Comment:
You could use the `# (1)` thing if you wanted

Reply:
Could do, although I consider it to be more useful for hiding larger amounts of text. Not quite worth it here.